### PR TITLE
Adding support for launching tasks (jobs) using Chronos

### LINKED ISSUE #5: Implement Mesos TaskLauncher
As a developer, I'd like to implement `TaskLauncher` contracts for mesos deployer.

**Acceptance:**
- `launch()` API allows launching Task applications
- `status()` API queries the current state of the Task application
- `delete()/destroy()` API removes the running Task application
- Unit tests included
- Docs included

### LINKED ISSUE #7: Add support for command-line arguments
As a developer, I want to be able to configure command line args as a separate set of properties to be passed to an app when it's deployed.

Similar to the Local SPI (_via: spring-cloud/spring-cloud-dataflow#597_), we would need this supported at the Mesos deployer SPI.

### LINKED ISSUE #20: Find or create a Java library for working with Chronos
We need a Java library for submitting and managing jobs (tasks) with Chronos, doesn't seem to be anything out there that's usable.

### PR DESCRIPTION
**Pull request by trisberg**
- updating the Chronos Java library
- adding support for command line args

Resolves #5
Resolves #7
Resolves #20

### CODE REVIEW

File: src/main/java/org/springframework/cloud/deployer/spi/mesos/chronos/ChronosTaskLauncherProperties.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+@ConfigurationProperties("spring.cloud.deployer.mesos.chronos")
+public class ChronosTaskLauncherProperties {
+
+	/**
+	 * The location of the Chronos REST endpoint.
+	 */
+	private String apiEndpoint = "http://m1.dcos/service/chronos";
+
+	/**
+	 * Secrets for a access a private registry to pull images.
```
Comment (ilayaperumalg) [inline src/main/java/org/springframework/cloud/deployer/spi/mesos/chronos/ChronosTaskLauncherProperties.java]: typo

File: src/main/java/org/springframework/cloud/deployer/spi/mesos/chronos/ChronosTaskLauncher.java [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+				job = j;
+			}
+		}
+		TaskStatus status = buildTaskStatus(properties, id, job, csv);
+		logger.debug(String.format("Status for task: %s is %s", id, status));
+
+		return status;
+	}
+
+	//	@Override
```
Comment (ilayaperumalg) [inline src/main/java/org/springframework/cloud/deployer/spi/mesos/chronos/ChronosTaskLauncher.java]: can be removed?

File: src/main/java/org/springframework/cloud/deployer/spi/mesos/chronos/ChronosTaskLauncher.java [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+	public void cleanup(String... id) {
+
```
Comment (ilayaperumalg) [inline src/main/java/org/springframework/cloud/deployer/spi/mesos/chronos/ChronosTaskLauncher.java]: extra line.

File: src/main/java/org/springframework/cloud/deployer/spi/mesos/chronos/ChronosTaskLauncher.java [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+			}
+		}
+		TaskStatus status = buildTaskStatus(properties, id, job, csv);
+		logger.debug(String.format("Status for task: %s is %s", id, status));
+
+		return status;
+	}
+
+	//	@Override
+	public void cleanup(String... id) {
```
Comment (ilayaperumalg) [inline src/main/java/org/springframework/cloud/deployer/spi/mesos/chronos/ChronosTaskLauncher.java:159]: We need to somehow use this method when destroying the tasks? Currently, the behaviour of task definition controller is to just delete the task definition repository entry. I think it should as well call `cleanup`. May be we need to have some contract in `TaskLauncher` for `cleanup`? \cc @trisberg @mminella @cppwfs

File: src/main/java/org/springframework/cloud/mesos/chronos/client/ChronosClient.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+		public void apply(RequestTemplate template) {
+			//template.header("Accept", "application/json");
+			template.header("Content-Type", "application/json");
+		}
+	}
+	
+	static class ChronosErrorDecoder implements ErrorDecoder {
+		@Override
+		public Exception decode(String methodKey, Response response) {
+			System.out.println(methodKey);
```
Comment (ilayaperumalg) [inline src/main/java/org/springframework/cloud/mesos/chronos/client/ChronosClient.java]: Are these intentional?

File: src/main/java/org/springframework/cloud/mesos/chronos/client/ChronosClient.java [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+	
+	public static Chronos getInstance(String endpoint) {
+		return getInstance(endpoint, null);
+	}
+
+	/**
+	 * The generalized version of the method that allows more in-depth customizations via
+	 * {@link RequestInterceptor}s.
+	 *
+	 * @param endpoint
```
Comment (ilayaperumalg) [inline src/main/java/org/springframework/cloud/mesos/chronos/client/ChronosClient.java]: Can be better formatted.

File: src/main/java/org/springframework/cloud/mesos/chronos/client/ChronosClient.java (hunk 5 above)
Comment (trisberg) [reply src/main/java/org/springframework/cloud/mesos/chronos/client/ChronosClient.java]: Don't think so ;)

**Comment by ilayaperumalg** [comment]
LGTM, merging

**Comment by ilayaperumalg** [comment]
Squashed the commits and merged as aff7ca1a9eeb909c7f6b936072ef470a762c210a

**Pull request closed without merging**